TTS/tts/utils/speakers.py

```diff
@@ -242,12 +289,16 @@ def compute_d_vector_from_clip(self, wav_file: Union[str, list]) -> list:
 
         def _compute(wav_file: str):
             waveform = self.speaker_encoder_ap.load_wav(wav_file, sr=self.speaker_encoder_ap.sample_rate)
-            spec = self.speaker_encoder_ap.melspectrogram(waveform)
-            spec = torch.from_numpy(spec.T)
+            if not self.speaker_encoder_config.model_params.get("use_torch_spec", False):
```

Comment:
This attribute has nothing to do here. It is again Monkey Patching

Reply:
Speaker Encoder should decide what to use internally

Reply:
I don't understand what you mean. Given that the speaker encoder doesn't have access to AP. So is here the right place to extract the mel-spectrogram. Do you suggest that we need to pass the AP to the Speaker encoder and extract the mel-spec in the Speaker encoder class?

This "IF" just controls the input. The new speaker encoder needs to receive a wave as input.  To remove this "IF" we need to pass both the mel_spec and waveform to the speaker encoder or pass the AP to the speaker encoder  ...